feat:사용자(회원+주최자) 이벤트 상세보기 기능

Yo0oN opened the pull request.

Yo0oN committed:
사용자(회원+주최자) 이벤트 상세보기 기능

[#27]과 같은 구조를 사용 예정이므로 해당 브랜치의 Controller, Service, DAO, DTO 참고하여 추가
Mapper 생성예정

[#36] 사용자(회원+주최자) 이벤트 상세보기 기능 추가

Yo0oN committed:
feat:사용자(회원+주최자) 이벤트 상세보기 기능

Mapper 추가
리턴값을 List<EventDTO>가 아닌 EventDTO로 수정

[#36]feat:사용자(회원+주최자) 이벤트 상세보기 기능

Jane096 commented on src/main/resources/mapper/EventsMapper.xml:
```diff
@@ -0,0 +1,17 @@
+            A.registerDate,
+            A.participants,
+            A.writer
+        FROM NULL events AS A
```
테이블에 Alias 값을 주신 이유가 있으신가요??

Yo0oN replied:
이부분도 사실 손에 익은대로 썼던거라 Alias를 주었던건데 수정하겠습니다..!

f-lab-dev commented on src/main/java/com/festa/controller/EventController.java:
```diff
@@ -0,0 +1,34 @@
+     */
+    @CheckLoginStatus(auth = UserLevel.USER)
+    @GetMapping("/{eventNo}")
+    public ResponseEntity<EventDTO> getInfoOfEvent(@PathVariable int eventNo) {
```
여기서는 ok만 리턴되므로 그냥 `EventDTO`를 리턴해줘도 될 것 같습니다~

Yo0oN replied:
네 수정하겠습니다!

Yo0oN replied:
넵 수정하겠습니다!

Yo0oN committed:
feat:사용자(회원+주최자) 이벤트 상세보기 기능

EventsMapper.xml에서 테이블이 하나임에도 Alias 사용한 부분 수정하여 Alias 제거

[#36] 사용자(회원+주최자) 이벤트 상세보기 기능 추가

Yo0oN committed:
feat:사용자(회원+주최자) 이벤트 상세보기 기능

EventController.java의 getInfoOfEvent 메서드의 리턴값을 Optional<EventDTO>로 수정

[#36] 사용자(회원+주최자) 이벤트 상세보기 기능 추가

f-lab-dev commented on src/main/java/com/festa/controller/EventController.java:
```diff
@@ -22,13 +23,15 @@
-    public ResponseEntity<EventDTO> getInfoOfEvent(@PathVariable int eventNo) {
-        EventDTO infoOfEvent = eventService.getInfoOfEvent(eventNo);
-        return ResponseEntity.ok(infoOfEvent);
+    public Optional<EventDTO> getInfoOfEvent(@PathVariable int eventNo) {
```
Controller 인데 `Optional`을 리턴하는 이유가 있으실까요?

Yo0oN replied:
이벤트를 조회하려 할 때 이벤트 조회를 실패하고 null이 반환되더라도 null을 반환해주지 않기 위해 Optional을 리턴했습니다.

f-lab-dev replied:
이벤트 조회가 실패하면 예외를 던져주거나 적절한 에러 status code를 던져주는게 낫지 않을까요?

Yo0oN replied:
에러를 던져주는건 안좋을거같아서 그냥 null이담긴 optional을 넘기려 했는데 에러 status code를 던져줘도 괜찮을까요?

f-lab-dev replied:
네 response body에는 무슨 에러인지 구체적으로 알려줄 수 있으면 그것도 좋을 것 같고요~

Yo0oN replied:
넵! 이벤트가 존재하지 않을 경우 IllegalArgumentException로 예외를 던져주도록 수정했습니다!

Yo0oN committed:
feat:사용자(회원+주최자) 이벤트 상세보기 기능

리턴값을 EventDTO로 수정, 이벤트 정보가 없을 경우 IllegalArgumentException 발생

[#36]feat:사용자(회원+주최자) 이벤트 상세보기 기능

f-lab-dev commented on src/main/java/com/festa/controller/EventController.java:
```diff
@@ -24,14 +20,16 @@
+    public EventDTO getInfoOfEvent(@PathVariable int eventNo) {
+        EventDTO infoOfEvent = eventService.getInfoOfEvent(eventNo);
+        if (infoOfEvent == null) {
+            throw new IllegalArgumentException("존재하지 않는 이벤트입니다.");
```
이 곳은 Controller이므로 예외를 던져주는것보다는 status code를 주는게 나을 것 같은데 예외를 던지는 것과 어떤 차이가 있을지 고민해보시면 좋을 것 같습니다

Yo0oN replied:
예외를 던지면 그곳에서 중단되고 status code가 500으로 처리되지만, 예외를 던지지 않고 status code를 주면 500 외에 각 상황에 맞는 status code를 주어 상황에 따라 다르게 대처할 수 있습니다!

f-lab-dev replied:
네 그것도 있지만 `Exception`을 던진다는 것이 어떤 의미를 갖는지도 고민해보시면 좋을 것 같습니다~

Yo0oN committed:
feat:사용자(회원+주최자) 이벤트 상세보기 기능

getInfoOfEvent() 리턴값 변경

[#36]feat:사용자(회원+주최자) 이벤트 상세보기 기능

Yo0oN committed:
Merge branch 'develop' into feature/#36
